myFitnessTrainer/src/components/GenerateNewPlanAlgoButton.js

```diff
@@ -0,0 +1,130 @@
+import { db } from "../../firebaseConfig";
+import { auth } from "../../firebaseConfig";
+import { getDocumentId, postDocument } from "../../databaseFunctions";
+
+/*
+    Pass in the following props as the below types:
+    - goal: Array<String>
+    - level: String
+    - duration: Int
+    - modifications: Array<String>
+*/
+const GenerateNewPlanAlgoButton = ({ goal, level, duration }) => {
+    // Test props for debugging
+    goal = goal ?? "strength";
+    level = level ?? "beginner";
+    duration = duration ?? 5;
+    // modifications = modifications ?? ["legs"];
+
+    const navigation = useNavigation();
+
+    /*
+    performs query to db for exercises collection based on goal and level survey result and returns an array of exercises
+    */
+    const getExercises = async (goal, level) => {
+        const q = query(
+            collection(db, "exercises"),
+            filterExerciseQuery(goal, level)
+        );
+        const querySnapshot = await getDocs(q);
+        const exercises = [];
+        querySnapshot.forEach((doc) => {
+            exercises.push(doc.data());
+        });
+        return exercises;
+    };
+
+    const createPlan = (exercises, duration) => {
+        const workoutPlan = [];
+
+        for (let i = 0; i < exercises.length && i < duration; i++) {
+            let randomIdx = Math.floor(Math.random() * exercises.length);
+            workoutPlan.push(exercises[randomIdx]);
+        }
+        return workoutPlan;
+    };
+
+    // TODO: Need to add ability to filter with modifications
+    const filterExerciseQuery = (goal, level) => {
+        switch (level) {
+            case "beginner":
+                return and(
+                    where("fitness_goal", "==", goal),
+                    where("fitness_level", "==", "beginner")
+                );
+            case "intermediate":
+                return and(
+                    where("fitness_goal", "==", goal),
+                    where("fitness_level", "==", "intermediate")
+                );
+            case "advanced":
+                return and(
+                    where("fitness_goal", "==", goal),
+                    where("fitness_level", "==", "advanced")
+                );
+            default:
+                break;
+        }
+    };
+    /* 
+        1. query all exercises with criteria goal, level, and modifications
+        2. create plan by selecting number of exercises == duration, random based from query 
+        3. add this plan to workoutplans collection
+    */
+    const handleCreateNewPlanAlgo = async () => {
+        const retrievedExercises = await getExercises(goal, level);
+        const workoutPlan = createPlan(retrievedExercises, duration);
+        const userId = await getDocumentId(
+            "users",
+            "email",
+            auth.currentUser.email
+        );
+
+        const postObject = {
+            workout_plan: workoutPlan,
```

Comment:
I'm going back & forth on whether to change this back to daily_exercises or if I like it called workout_plan

Reply:
I changed the field in the Firestore document to daily_exercises, but the Firestore collection name is still workout_plans. Let me know if we prefer to change the collection name too.

Reply:
I think we can keep the collection with the name workout_plans